sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -0,0 +1,245 @@
+package io.sentry;
+
+import io.sentry.protocol.Contexts;
+import io.sentry.protocol.Request;
+import io.sentry.protocol.SentryId;
+import io.sentry.protocol.SentryTransaction;
+import io.sentry.util.Objects;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicReference;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class SentryTracer implements ITransaction {
+  private final @NotNull SentryId eventId = new SentryId();
+  private final @NotNull Span root;
+  private final @NotNull List<Span> children = new CopyOnWriteArrayList<>();
+  private final @NotNull IHub hub;
+
+  public SentryTracer(final @NotNull TransactionContext context, final @NotNull IHub hub) {
+    Objects.requireNonNull(context, "context is required");
+    Objects.requireNonNull(hub, "hub is required");
+    this.root = new Span(context, this, hub);
+    this.root.setTag(ISpan.NAME_TAG, context.getName());
```

Comment:
`setTag` was added to Span just so that we can do this, right?
Couldn't we just have a string field for `name` and an API to set TransationName on `ITransaction` instead?

Reply:
This tag is a leftover from the refactoring to drop `ITransaction` completely.

Tags were on span before, but I agree that we can create separate field `name` on SentryTracer.

Done.